internal/pkg/cosign/sign.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cosign
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/mutate"
+	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
+	"github.com/sigstore/cosign/pkg/oci/static"
+	"github.com/sigstore/sigstore/pkg/signature"
+	signatureoptions "github.com/sigstore/sigstore/pkg/signature/options"
+)
+
+type SigningRequest struct {
+	SignaturePayload []byte
+	SignedEntity     oci.SignedEntity
+}
+
+type SigningResults struct {
+	SignedPayload []byte
+	Signature     []byte
+	Pub           crypto.PublicKey
+	Cert, Chain   []byte
+	Bundle        *oci.Bundle
+
+	OCISignature oci.Signature
+	SignedEntity oci.SignedEntity
+}
```

Comment:
The SignedEntity is what's passed in, and virtually all of the stuff on top is accessible from the `oci.Signature`, so I think the result should just be `oci.Signature`.

Reply:
So it seems like this result struct is built up across multiple implementations that must be composed a particular way to work, which isn't really the goal of composition. 🤔 

Given the pieces I'm seeing I think the interface should just be:
```go
type Signer interface {
	Sign(ctx context.Context, payload []byte) (oci.Signature, error)
}
```

Reply:
roger. The only thing not captured by `oci.Signature` was the signature's public key